python/ray/_raylet.pyx

```diff
@@ -816,10 +857,29 @@ cdef execute_task(
                                               error, proctitle=title,
                                               actor_repr=actor_repr)
             errors = []
-            for _ in range(c_return_ids.size()):
+            for _ in range(returns[0].size()):
                 errors.append(failure_object)
             core_worker.store_task_outputs(
-                worker, errors, c_return_ids, returns)
+                caller_address, worker, errors,
+                False,  # is_dynamic
+                CObjectID.Nil(),
+                returns)
+            if dynamic_returns != NULL:
```

Comment:
I'm a bit confused. Does `dynamic_returns == NULL` mean that we are doing reconstruction? From line 815, seems it's not.

Reply:
It means that the caller set `num_returns="dynamic"`.